zulip_bots/zulip_bots/lib.py

```diff
@@ -239,9 +239,12 @@ def extract_query_without_mention(message: Dict[str, Any], client: ExternalBotHa
     the stripped message with the bot's @mention removed.  Otherwise, it returns None.
     """
     mention = '@**' + client.full_name + '**'
-    if not message['content'].startswith(mention):
-        return None
-    return message['content'][len(mention):].lstrip()
+    if message['content'].startswith(mention):
+        return message['content'][len(mention):].lstrip()
+    extended_mention = '@**' + client.full_name + '|' + str(client.user_id) + '**'
+    if message['content'].startswith(extended_mention):
+        return message['content'][len(extended_mention):].lstrip()
+    return None
```

Comment:
I think it'd be better to do an `if/elif` check, starting with checking for `extended_mention`, so that it's clearer that this won't remove 2 mentions from a message (logically).

I think there might also be a copy of similar logic in the `outgoing_webhook` part of the server repo.

Reply:
I figured using an early return should make that clear, but I'll make that change.